benchexec/tools/ultimate.py

```diff
@@ -68,30 +66,32 @@ def __init__(self):
         self.java = None
 
     @functools.lru_cache()
-    def executable(self):
-        exe = util.find_executable("Ultimate.py")
-        if exe:
-            for (_dirpath, dirnames, filenames) in os.walk(os.path.dirname(exe)):
-                if "Ultimate" in filenames and "plugins" in dirnames:
+    def executable(self, tool_locator):
+        exe = None
+        try:
+            exe = tool_locator.find_executable("Ultimate.py")
+            for (_, dir_names, file_names) in os.walk(os.path.dirname(exe)):
+                if "Ultimate" in file_names and "plugins" in dir_names:
                     return exe
                 break
+        except benchexec.tools.template.ToolNotFoundException:
+            pass
 
         # possibly another Ultimate.py was found or not found at all, check in the current dir
```

Comment:
Note that tool-info modules based on `BaseTool2` should not look for the tool in the current directory automatically anymore. `ToolLocator` will search in the current directory if and only if this is desired according to the BenchExec configuration.

And AFAIS this should mean that the whole method can and should be simplified to a simple call to `tool_locator.find_executable` as in most other tool-info modules.